internal/connector/connector.go

```diff
@@ -98,123 +101,79 @@ func (j *jwkCache) getJWK() (*jose.JSONWebKey, error) {
 	return &response.Keys[0], nil
 }
 
-var JWKCacheRefresh = 5 * time.Minute
-
-type BearerTransport struct {
-	Token     string
-	Transport http.RoundTripper
-}
-
-func (b *BearerTransport) RoundTrip(req *http.Request) (*http.Response, error) {
-	if b.Token != "" {
-		req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", b.Token))
+func newAuthenticator(url string, options Options) *authenticator {
```

Comment:
I created a constructor for this to remove some of the logic from the large `Run` function.